packages/klighd-interactive/src/tree/constraint-util.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { SModelElement } from 'sprotty';
+import { Action } from 'sprotty-protocol';
+import { RefreshLayoutAction } from '../actions';
+import { Direction, KEdge, KNode } from '../constraint-classes';
+import { TreeSetPositionConstraintAction } from './actions';
+
+/**
+ * Calculates dot product of two vectors of size 2.
+ * This is used for directional vectors.
+ * 
+ * @param vec1 First vector.
+ * @param vec2 Second vector.
+ * @returns The dot product.
+ */
+export function dotProduct(vector1: [number, number], vector2: [number, number]): number {
```

Comment:
this seems like a very generic maths util function that maybe should be in more central place or should use an actual vector class

Reply:
All other interactive strategies could potentially use this method to avoid big switch case constructs for all directions. I will move it to the general constraint-utils file.